main.nf

```diff
@@ -1,130 +1,575 @@
+        </dl>
+    """.stripIndent()
+
+   return yaml_file
+}
+
 
 Channel
-    .value([p.id, p.fastqs])
-    .into { supernova_input; longranger_input }
+    .from(samples)
+    .set { supernova_input }
+
+if (params.full_output) {
+    process supernova_full {
+        tag "${id}"
+        label "supernova"
+        publishDir "${params.outdir}/supernova/", mode: 'copy'
+
+        input:
+        set val(id), val(fastqs), val(tenx_options), val(supernova_options) from supernova_input
+
+
+        output:
+        set val(id), file("${id}/*") into supernova_results, supernova_results2
+
+        script:
+        """
+        supernova run --id=${id} --fastqs=${fastqs} ${tenx_options} ${supernova_options}
+        """
+    }
 
-process supernova {
+} else {
+    process supernova {
+        tag "${id}"
+        label "supernova"
+        publishDir "${params.outdir}/supernova/", mode: 'copy'
+
+        input:
+        set val(id), val(fastqs), val(tenx_options), val(supernova_options) from supernova_input
+
+        output:
+        set val(id), file("${id}_supernova") into supernova_results, supernova_results2
+
+        script:
+        """
+        supernova run --id=${id} --fastqs=${fastqs} ${tenx_options} ${supernova_options}
+        rsync -rav --include="_*" --include="*.tgz" --include="outs/" --include="outs/*.*"  --include="assembly/" --include="stats/***" --include="logs/***" --include="a.base/" --include="a.base/" --include="a.hbx" --include="a.inv" --include="final/***" --include="gang" --include="micro"  --include="a.hbx" --include="a.inv" --include="final/***" --exclude="*" "${id}/" ${id}_supernova
+        """
+    }
+
+}
+
+process supernova_mkoutput {
     tag "${id}"
-    publishDir "${p.outdir}/supernova/", mode: 'copy'
+    label "supernova"
+    publishDir "${params.outdir}/assemblies/", mode: 'copy'
 
     input:
-    set val(id), val(fastqs) from supernova_input
+    set val(id), file(supernova_folder) from supernova_results
 
     output:
-    set val(id), file("${id}") into supernova_results
+    set val(id), file("${id}.fasta") into supernova_asm1, supernova_asm2
+    file "${id}.phased.fasta"
 
     script:
-    def mem = Math.round(task.memory.toBytes() / 1024 / 1024 / 1024)
     """
-    supernova run --id=${id}_stage --fastqs=${fastqs} --localcores=${task.cpus} --localmem=${mem} ${supernova_optional()}
-    rsync -rav --include="_*" --include="*.tgz" --include="outs/" --include="outs/*.*"  --include="assembly/" --include="stats/***" --include="logs/***" --include="a.base/" --include="a.hbx" --include="a.inv" --include="final/***" --exclude="*" "${id}_stage/" ${id}
+    supernova mkoutput --asmdir=${id}_supernova/outs/assembly --outprefix=${id} --style=pseudohap --minsize=${params.minsize} --nozip
+    supernova mkoutput --asmdir=${id}_supernova/outs/assembly --outprefix=${id}.phased --style=megabubbles --minsize=${params.minsize} --nozip
```

Comment:
How long running is this? Is it worth breaking the second one out into its own process for better parallelisation?

Reply:
For 95% of all assemblies I would imagine this runs for 3-5 mins per `mkoutput` command. I don't think it needs to be parallelized

Reply:
Ok 👍